### PR TITLE
Add function to authorize user to register patient

### LINKED ISSUE #8988: Infinite Rendering and Redirection to 'not-found' Page When Adding Patient from Non-Home Facility
**Describe the bug**  
When logged in as "Doctor Dev," attempting to add a new patient from the "View Facility" section in a facility that is not the user's home facility causes an infinite rendering loop. This results in a redirection to the "not-found" page.

**To Reproduce**  
1. Go to **/facility**
2. Click on **View Facility**
3. In any facility that is not the home facility, click on **Add Details of Patient**
4. Observe the error (infinite rendering and redirection to "not-found" page)

**Expected behavior**  
The system should display a notification with an appropriate message explaining the restriction and redirect the user back to the home page instead of triggering an error.

**Video**  
[Watch Video](https://github.com/user-attachments/assets/5bf92849-d240-4a8b-8dc3-1166f430781d)

**Desktop (please complete the following information):**
- **OS**: Windows
- **Browser**: Chrome
- **Version**: 130.0.6723.92

### PR DESCRIPTION
**Pull request by JavidSumra**
## Proposed Changes

- Fixes #8988 
- Updated `isAuthorizedToRegisterPatient` function to handle authorization checks when adding a new patient from the "View Facility" page.
- Implemented a check to prevent infinite rendering and redirect loops by displaying a notification with an appropriate message.
- Redirects unauthorized users back to the home page if they attempt to add a patient in a facility not designated as their home facility.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug fix or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that all UI text is stored in I18n files.
- [ ] Prepare and attach a screenshot or demo video for the changelog entry to the issue.
- [ ] Request peer reviews.
- [ ] Complete QA.

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic rendering for the "Add Patient" button in the FacilityHome component, ensuring only authorized users can access it.
	- Enhanced user authentication logic for patient registration, streamlining the process based on user roles and facility context.

- **Bug Fixes**
	- Improved control flow for user permissions, ensuring accurate access rights for patient registration.

### CODE REVIEW

**Comment by nihal467** [comment]
@JavidSumra iam seeing changes suggested by ai, look into that as well

**Comment by nihal467** [comment]
@JavidSumra 
image

we already have this auth defined in the code base, so use that rather than duplicating it :)

**Comment by JavidSumra** [comment]
Hey, @nihal467 requested changes are done can you please review this PR?

**Comment by nihal467** [comment]
LGTM

File: src/components/Facility/FacilityHome.tsx [hunk 1]
```diff
@@ -458,17 +460,19 @@ export const FacilityHome = ({ facilityId }: Props) => {
...
-                ghost
-                border
-                className="mt-2 flex w-full flex-row justify-center md:w-auto"
-                onClick={() => navigate(`/facility/${facilityId}/patient`)}
-                authorizeFor={NonReadOnlyUsers}
-              >
-                <CareIcon icon="l-plus" className="text-lg" />
-                <span className="text-sm">{t("add_details_of_patient")}</span>
-              </ButtonV2>
+              {PatientRegisterAuth(authUser, facilityData, facilityId) && (
```
Comment (rithviknishad) [inline src/components/Facility/FacilityHome.tsx]:
```suggestion
              {patientRegisterAuth(authUser, facilityData, facilityId) && (
```

File: src/components/Facility/FacilityHome.tsx [hunk 2]
```diff
@@ -48,6 +48,8 @@ import uploadFile from "@/Utils/request/uploadFile";
 import useQuery from "@/Utils/request/useQuery";
 import { sleep } from "@/Utils/utils";
 
+import { PatientRegisterAuth } from "../Patient/PatientRegister";
```
Comment (rithviknishad) [inline src/components/Facility/FacilityHome.tsx]:
```suggestion
import { patientRegisterAuth } from "../Patient/PatientRegister";
```

File: src/components/Patient/PatientRegister.tsx [hunk 3]
```diff
@@ -814,31 +816,12 @@ export const PatientRegister = (props: PatientRegisterProps) => {
...
-
-    return false;
-  };
-
-  if (!isLoading && facilityId && facilityObject && !PatientRegisterAuth()) {
+  if (
+    !isLoading &&
+    facilityId &&
+    facilityObject &&
+    !PatientRegisterAuth(authUser, facilityObject, facilityId)
```
Comment (rithviknishad) [inline src/components/Patient/PatientRegister.tsx]:
```suggestion
    !patientRegisterAuth(authUser, facilityObject, facilityId)
```

File: src/components/Patient/PatientRegister.tsx [hunk 4]
```diff
@@ -1713,3 +1696,31 @@ export const PatientRegister = (props: PatientRegisterProps) => {
     </Form>
   );
 };
+
+export function PatientRegisterAuth(
```
Comment (rithviknishad) [inline src/components/Patient/PatientRegister.tsx]:
```suggestion
export function patientRegisterAuth(
```

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**